src/mds/MDBalancer.cc

```diff
@@ -150,11 +150,20 @@ void MDBalancer::handle_export_pins(void)
 
   set<CDir *> authsubs;
   mds->mdcache->get_auth_subtrees(authsubs);
+
+  if (authsubs.size() > AUTH_TREES_THRESHOLD &&
+      !g_conf()->subsys.should_gather<ceph_subsys_mds, 25>()) {
+    dout(15) << "number of auth trees = " << authsubs.size() << "; not "
+		"printing auth trees" << dendl;
+    return;
+  }
```

Comment:
````
2019-03-01 19:39:07.919 1aa69700 20 mds.6.bal export_pin_queue size=0
2019-03-01 19:39:07.919 1aa69700 15 mds.6.bal number of auth trees = 34; not printing auth trees
2019-03-01 19:39:07.919 1aa69700 15 mds.6.bal tick last_sample now 3644.81s
2019-03-01 19:39:07.919 1aa69700 15 mds.6.cache show_subtrees
2019-03-01 19:39:07.920 1aa69700 10 mds.6.cache |__________ 6    auth [dir 0x106 ~mds6/ [2,head] auth v=11 cv=1/1 dir_auth=6 state=1610612737|complete f(v0 10=0+10) n(v1 rc2019-03-01 19:37:27.978749 b4194304 11=1+10)/n(v1 rc2019-03-01 19:36:30.306667 b1973 11=1+10) hs=10+0,ss=0+0 dirty=1 | child=1 subtree=1 dirty=1 authpin=0 0x1b381170]
````

From: /ceph/teuthology-archive/pdonnell-2019-03-01_18:13:24-multimds-wip-pdonnell-testing-20190301.150213-distro-basic-smithi/3654417/remote/smithi178/log/ceph-mds.e.log.gz

This isn't right. You are skipping the debug prints in the following lines but you're also skipping the call to `export_dir`.

Reply:
Fixed.